org.eclipse.draw2d/src/org/eclipse/draw2d/geometry/Translatable.java

```diff
@@ -15,6 +15,38 @@
  * horizontally.
  */
 public interface Translatable {
+	/**
+	 * Translates this object horizontally by <code>point.x</code> and vertically by
+	 * <code>point.y</code>.
+	 * 
+	 * @param point Point which provides translation information
+	 * @since 3.13
+	 */
+	default void performTranslate(Point point) {
+		performTranslate(point.x, point.y);
+	}
+
+	/**
+	 * Translates this object horizontally by <code>dimension.width</code> and
+	 * vertically by <code>dimension.height</code>.
+	 * 
+	 * @param dimension Dimension which provides translation information
+	 * @since 3.13
+	 */
+	default void performTranslate(Dimension dimension) {
+		performTranslate(dimension.width, dimension.height);
+	}
+
+	/**
+	 * Translates this object horizontally by <code>insets.left</code> and
+	 * vertically by <code>insets.top</code>.
+	 * 
+	 * @param insets Insets which provides translation information
+	 * @since 3.13
+	 */
+	default void performTranslate(Insets insets) {
```

Comment:
I'm not sure about the translate with an Insets object. An Insets has in four values why take left and top? Or should this method have then a different name?

Reply:
> why take left and top

When it comes to painting Figures, all coordinates have to be relative to the top-left corner of the canvas.

So if you want to e.g. only draw the client area of a Figure, you need to convert all coordinates, relative to the Figure (or their parent Figure), to their absolute coordinates of the panel. Meaning the bounds of the Figure as well as its Insets.

So in a way, the (left, top) coordinates of an Insets correspond to the (x, y) coordinates of a Point, which is why I treat them in the same way.

Reply:
Thx for the explanation. I think I was to tired when I reviewed the code yesterday evening. With your explanation it totally makes sense. Sorry for the disturbance.